### PR TITLE
metabase: update to 0.47.1

### PR DESCRIPTION
**Pull request by slashapp**
###### Type(s)

- [ ] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS 13.5.2 22G91 arm64
Xcode 14.3.1 14E300c

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?
- [x] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

### CODE REVIEW

File: databases/metabase/Portfile [hunk 1]
```diff
@@ -21,12 +21,13 @@ supported_archs     noarch
 license             MIT
 
 master_sites        https://downloads.metabase.com/v${version}/
+dist_subdir         ${name}/${version}
```
Comment (reneeotten) [inline databases/metabase/Portfile:24]: why is this needed; does the default not work?

Comment (slashapp) [reply databases/metabase/Portfile:24]: The distfiles name is metabase.jar and there is no version name, when installing new version with the port install -b option, the old version matebase.tbz2 on the macports will cause the checksum failed.

This issue came up in the jenkins port, fixed the same way in this port.

ref -> https://github.com/macports/macports-ports/pull/20110#issuecomment-1692766631

**Review by mascguy — APPROVED** [review]
